### DISCUSSION TITLE
Esp32 Status error

### DISCUSSION THREAD

**Discussion by ocimjr** [corrados/edrumulus#83] [General]
Hi,

I started to play with this project. Be patient I'm a beginner in the world of electronic.
After put a test circuit see image:
![IMG_20230811_115707628_HDR](https://github.com/corrados/edrumulus/assets/37626833/23325b1b-cc4f-49d7-894c-c57e7085c0d3)

I notice some phantom triggers. Investigating I saw that the flag status_is_error is always true.

Log : 
After flash

12:08:59.525 -> status_is_error 0
12:09:01.997 -> Sample rate 3916.8471680
12:09:01.997 -> Max sample error Hz: 100
12:09:01.997 -> Detected : 4083.15
12:09:01.997 -> serial_print :  (pad)i : 0 (input )j:  0 -> 1931	1932.45	 (pad)i : 0 (input )j:  1 -> 1931	1932.66	
12:09:02.031 -> serial_print2 : -1.45	-1.66	

Then 

12:09:03.266 -> status_is_error 1
12:09:03.266 -> Sample rate 7805.6582031
12:09:03.266 -> Max sample error Hz: 100
12:09:03.266 -> Detected : 194.34
12:09:03.297 -> serial_print :  (pad)i : 0 (input )j:  0 -> 1934	1932.48	 (pad)i : 0 (input )j:  1 -> 1934	1932.59	
12:09:03.297 -> serial_print2 : 1.52	1.41
12:09:04.548 -> status_is_error 1
12:09:04.548 -> Sample rate 7805.6396484
12:09:04.548 -> Max sample error Hz: 100
12:09:04.548 -> Detected : 194.36
12:09:04.580 -> serial_print :  (pad)i : 0 (input )j:  0 -> 1931	1932.53	 (pad)i : 0 (input )j:  1 -> 1930	1932.56	
12:09:04.580 -> serial_print2 : -1.53	-2.56	
12:09:05.833 -> status_is_error 1
12:09:05.833 -> Sample rate 7805.6396484
12:09:05.833 -> Max sample error Hz: 100
12:09:05.833 -> Detected : 194.36
12:09:05.833 -> serial_print :  (pad)i : 0 (input )j:  0 -> 1931	1932.68	 (pad)i : 0 (input )j:  1 -> 1936	1932.63	
12:09:05.865 -> serial_print2 : -1.68	3.37

I have only one pad connect pd-80r

Probably I did some wrong.  

Thanks

**Comment by corrados** [comment]
Sampling rate 7800 Hz should be sufficient to geht good triggering. When you hit the pad, do you get correct MIDI notes?

**Reply by ocimjr** [reply]
Yes notes are correct. Triggering seems fine. Correct notes and velocity. 
But since, I had some phantom notes I started to investigate and find this status error

**Reply by corrados** [reply]
Try to increase the threshold. Or try to use a different USB port.

**Reply by ocimjr** [reply]
Ok, but this will make status to not be anymore in error?

**Reply by corrados** [reply]
You can ignore this status error for now.

**Reply by ocimjr** [reply]
Ok thanks for your help

**Reply by ocimjr** [reply]
I was playing to see the read of ADC and when de 3.3v circuit is connect to esp32 the reading of snare and rim are in the region of 1900 without hitting the drums.
 If disconnect the 3.3v circuit the reading drops to 0.

I was wondering which values should be the ADC when I'm not hitting the pad? Do you know?

Thanks again

**Reply by corrados** [reply]
1900 is correct.

**Reply by ocimjr** [reply]
I'm almost starting a real prototype and I'm wondering if you use a shield for your esp32 or juste the board?

**Reply by corrados** [reply]
As you can see in the pictures of the [change log file](https://github.com/corrados/edrumulus/blob/main/doc/ChangeLog.md), I am currently not using shielding. I think it would be a good idea to use shielding but it seems drum modules by Roland don't do it, e.g., [as seen in the TD-6 module](https://i0.wp.com/wphost.spider-e.com/wp-content/uploads/2015/02/mainboard.png) or the TD-10 module:
![grafik](https://github.com/corrados/edrumulus/assets/46655886/8ed9b4e8-458c-404e-aa34-815a12010a03)
